Release 1.0.0-20180326-1

stephenrichards opened the pull request.

aliuk2012 commented on app/policies/case_policy.rb:
```diff
@@ -23,6 +23,13 @@ def can_accept_or_reject_case?
+  def can_remove_attachment?
+    case self.case.current_state
+    when 'drafting' then self.case.drafter == user
+    when 'responded' then user.assigner?
```
Do we know whether assigners should be able to remove uploaded responses?

stephenrichards replied:
The answer is: NO

stephenrichards replied:
IN fact, NO-ONE should be able to remove an uploaded file once the case has been marked as reponded

misaka commented:
Just to comment on the specs, in case it's not obvious I've done a couple changes to them:

- broken out the `uploaded_files_spec` into three specs:
  - `viewing_response_details_spec`
  - `downloading_response_spec`
  - `removing_response_spec`
- reorganised the structure of many of the specs in these files so that the scenario isn't too honking huge and to try to remove a little bit of the duplication ... not because tests need to be totally DRY (they don't) ... but to try to reduce some of the noise in the spec features. This is the structure I used:
  - context of logged in as assigner/drafter
    - context of case state (drafting, responded, etc)
      - scenario

aliuk2012 commented on app/policies/case_policy.rb:
```diff
@@ -23,6 +23,13 @@ def can_accept_or_reject_case?
     self.case.awaiting_responder? && self.case.drafter == user
   end
 
+  def can_remove_attachment?
```
We need an "can_remove_attachment" translation in the locales file.

misaka replied:
Added.

Commit:
CT-1568 adds tests for permitted events

Commit:
CT-1568 events for trigger cases awaiting responder config state machine

Commit:
CT-1568 events for trigger cases awaiting responder config state machine

Commit:
CT-1568 WIP drafting events trigger case

Commit:
CT-1568 WIP drafting events trigger case

Commit:
WIP

Commit:
WIP

stephenrichards committed:
CT-1568 Fixed extend_for_pit and specs

stephenrichards committed:
CT-1568 Fixed extend_for_pit and specs

stephenrichards committed:
CT-1568 Fixed policies for extend for pit

stephenrichards committed:
CT-1568 Rebase develop

stephenrichards committed:
CT-1568 Rebase develop

stephenrichards committed:
CT-1568 Cocde reformatting

stephenrichards committed:
CT-1568 Cocde reformatting

Commit:
CT-1612 adds permitted events for trigger pending dacu_clearance state_machine

stephenrichards committed:
CT-1568 PR Review issues

stephenrichards committed:
CT-1568 fix spec with new method signature

Commit:
CT-1612 creates full_approval workflow

Commit:
CT-1612 updates workflows in the factories

misaka committed:
CT-1568 add CasesController#remove_clearance for clarity

This action doesn't need to do anything, but it'd be nice if it was there, so
that it's less confusing when trying to understand the system, and if it
authorized to ensure that users don't even think they can remove clearance if
they happen to get to load this page somehow.

Commit:
CT-1634 adds predicate for preventing emails

misaka committed:
CT-1635 add missing events for approvers on unassigned cases

Some of the missing events were unflag_for_clearance and
unaccept_approver_assignment, which were blocking the undo for Take Case On

Commit:
CT-1634 fixes indentation of private

beccapearce committed:
Merge pull request #106 from ministryofjustice/ct-1634-prevent-emails-on-own-actions

CT-1634 adds predicate for preventing emails

Commit:
CT-1612 updates request amends tests to pending private clearance cases

misaka committed:
CT-1635 stop updating log/parallel_runtime_rspec.log

Let's just stop updating this file automatically, this should properly stop the
conflicts we've been having.

misaka committed:
CT-1568 adapt unflag_for_clearance to new state machine

Both the controller and the state machine used to call the policy
BasePolicy#can_unflag_for_clearance?. This moves the policy logic into
Workflows::Predicates, points the state machine to use that, and changes the
policy to check whether the state machine can trigger the event.

Commit:
CT-1612 adds full_approval workflow to state_machines_and_workflows_spec

misaka committed:
CT-1568 add disclosure_specialist_coworker to predicate specs

misaka committed:
Merge pull request #107 from ministryofjustice/ct-1635-allow-ds-to-undo-taking-case-on

CT-1635 add missing events for approvers on unassigned cases

misaka committed:
CT-1568 add missing events to permitted_events for approvers

misaka committed:
CT-1568 disable Rubocop method length check for trigger_event

This method should be re-factored to extract code common to can_trigger_event?()
and possibly event_present_and_triggerable?().

Commit:
CT-1634 passes user to after_transtion methods

Commit:
CT-1634 reomves unescessary variable

misaka committed:
Merge pull request #105 from ministryofjustice/ct-1568-trigger-config-state-machine

CT-1568 trigger config state machine for awaiting responder and drafting states

Commit:
CT-1612 adds next step info to config state machine

Commit:
CT-1612 fixes permitted events specs

stephenrichards committed:
Rubocop fixes

Commit:
WIP

Commit:
CT-1612 removes approve and bypass from trigger cases

Commit:
CT-1612 moves add_message policy in statemachine to use predicates

aliuk2012 committed:
CT-1632 improves error styling and message in dropzone

Commit:
CT-1634 adds a test for not calling mailer

Commit:
CT-1612 WIP

stephenrichards committed:
CT-1612 Fix specs

Commit:
CT-1612 prevents response uploads before escalation deadline

stephenrichards committed:
CT-1612 Migration to update workflow to full approval

stephenrichards committed:
CT-1658 Ensure messages added with uploaded files are stored

beccapearce committed:
Merge pull request #108 from ministryofjustice/ct-1634-prevent-emails-on-own-actions

CT-1634 passes user to after_transtion methods

stephenrichards committed:
Merge pull request #110 from ministryofjustice/ct-1568-message-upload-file-bug

CT-1658 Ensure messages added with uploaded files are stored

Commit:
CT-1612 update migration

Commit:
CT-1612 WIP

Commit:
CT-1612 updates specs to work with new state machine

stephenrichards committed:
CT-1612 Fix specs

Commit:
CT-1612 WIP

Commit:
swtiches workflow for take on for approval event

Commit:
CT1612 adds policy check for add respondes

stephenrichards committed:
CT-1612 refactor Machine#next_state_for_event

aliuk2012 committed:
CT-1644 updates loofah from 2.0.3 to 2.2.1

stephenrichards committed:
CT-1612 Specs and state machine transitions for unflag and unaccept

stephenrichards committed:
CT-1612 Remove unussed files

aliuk2012 committed:
CT-1600 requires missing CTS constants file

This fixes up the basic FOI case creation but the page still needs
some work when it comes to SARs like missing fields on the page and the
odd FOI case creation issue.

At least its not just an error page you see.

stephenrichards committed:
CT-1612 Fix PR issues

stephenrichards committed:
CT-1612 Make migrations usable even if class methods are removed

stephenrichards committed:
CT-1612 remove duplicate clone

stephenrichards committed:
Merge pull request #111 from ministryofjustice/ct-1612-foi-trigger-pending-dacu-clearance-state-machine

Ct 1612 foi trigger pending dacu clearance state machine

misaka committed:
Merge pull request #113 from ministryofjustice/ct-1600-admin-console

CT-1600 requires missing CTS constants file
